### DISCUSSION TITLE
OpenSpiel makefile

### DISCUSSION THREAD

**Discussion by LucasCelestinoSE** [deepmind/open_spiel#1188] [Q&A] [answered]
I've been studying cmake and I'm trying to create a cmake to build a game apart from the original build in the repository.

But there are some files and directories that are core even before building the examples or games.

I have been reading CmakeList which is inside the directory
openspiel
../openspiel

And I found the following supposed core_objetcs (which contains all the core codes)
And color directories.

I tested but was unable to build, giving some errors, are there any other CmakeLists with more color files/directories that I ended up not paying attention to?

**Comment by lanctot** [comment, marked as answer]
Hi @LucasCelestinoSE, 

Yes it would be difficult to pull a game out on its own because several of them depend on other things in the code base, but mainly, the core API. 

The best way to do this would be to, first, [build OpenSpiel as a C++ library](https://github.com/google-deepmind/open_spiel/blob/master/docs/library.md). Then have a separate project with its own Makefile (or CMakeLists.txt) that links against the OpenSpiel shared library. You can copy over the game that you want into your project, modify it however you like, give it a different name, and then load it under the new name. 

I hope this helps. If not, please elaborate on what you'd like to do in more detail.
